src/display/network.js

```diff
@@ -420,6 +430,14 @@ class PDFNetworkStreamRangeRequestReader {
     this.onClosed = null;
   }
 
+  _onHeadersReceived() {
+    this.#responseOrigin.resolve(
```

Comment:
Again this stage may never be reached. Shouldn't we resolve (or reject) earlier to avoid requests getting stuck?

Reply:
This would be similar to the situation outlined for `src/display/fetch_stream.js` above.